src/Compilers/CSharp/Portable/Symbols/LocalDeclarationKind.cs

```diff
@@ -49,5 +49,16 @@ internal enum LocalDeclarationKind : byte
         /// The variable that captures the result of a pattern matching operation like "i" in "expr is int i"
         /// </summary>
         PatternVariable,
+
+        /// <summary>
+        /// User variable declared by a declaration expression in the left-hand-side of a deconstruction assignment.
+        /// </summary>
+        Deconstruction,
+
+        /// <summary>
+        /// User variable declared by a declaration expression that is not a deconstruction. This occurs for an
+        /// out variable declaration, and as a result of error recovery in incorrect code.
+        /// </summary>
+        DeclarationExpression,
```

Comment:
`DeclarationExpression` only represents declaration expressions, period. But declaration expressions can appear in source in error scenarios, because that is the way we've elected to factor the language.